bond/bond_test.go

```diff
@@ -159,231 +146,335 @@ var _ = Describe("bond plugin", func() {
 				args.ContainerID, "", func() error { return cmdDel(args) })
 			Expect(err).NotTo(HaveOccurred())
 
-			return nil
 		})
-		Expect(err).NotTo(HaveOccurred())
-	})
 
-	It("verifies the del command does not fail when a device (in container) assigned to the bond has been deleted", func() {
-		args := &skel.CmdArgs{
-			ContainerID: "dummy",
-			Netns:       podNS.Path(),
-			IfName:      IfName,
-			StdinData:   []byte(fmt.Sprintf(Config, "1.0.0", ActiveBackupMode)),
-		}
-
-		err := podNS.Do(func(ns.NetNS) error {
-			defer GinkgoRecover()
+		It("verifies the del command does not fail when a device (in container) assigned to the bond has been deleted", func() {
+			args := &skel.CmdArgs{
+				ContainerID: "dummy",
+				Netns:       podNS.Path(),
+				IfName:      IfName,
+				StdinData:   []byte(fmt.Sprintf(Config, "1.0.0", ActiveBackupMode, strconv.FormatBool(linksInContainer), strconv.Itoa(DefaultMTU))),
+			}
 
 			By("adding a bond interface")
 			_, _, err := testutils.CmdAddWithArgs(args, func() error {
 				return cmdAdd(args)
 			})
 			Expect(err).NotTo(HaveOccurred())
 
-			By("deleting a slave interface")
-			slave, err := netlink.LinkByName(Slave1)
-			Expect(err).NotTo(HaveOccurred())
-			err = netlink.LinkDel(slave)
+			err = podNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+
+				By("deleting a slave interface")
+				slave, err := netlink.LinkByName(Slave1)
+				Expect(err).NotTo(HaveOccurred())
+				err = netlink.LinkDel(slave)
+				Expect(err).NotTo(HaveOccurred())
+				return nil
+			})
+
 			Expect(err).NotTo(HaveOccurred())
 
 			By("deleting the bond interface")
 			err = testutils.CmdDel(podNS.Path(),
 				args.ContainerID, "", func() error { return cmdDel(args) })
 			Expect(err).NotTo(HaveOccurred())
 
-			return nil
 		})
-		Expect(err).NotTo(HaveOccurred())
-	})
 
-	It("verifies the plugin returns correct results for CNI versions 0.3.0, 0.3.1, 0.4.0", func() {
-		for _, version := range []string{"0.3.0", "0.3.1", "0.4.0"} {
+		DescribeTable("verifies the plugin returns correct results for supported tested versions", func(version string) {
 			args := &skel.CmdArgs{
```

Comment:
replace args.StdinData = []byte(fmt.Sprintf(Config, version, ActiveBackupMode, strconv.FormatBool(linksInContainer), strconv.Itoa(DefaultMTU))) accordingly after moving args to beforeeach

Reply:
fixed